exporter/splunkhecexporter/hec_worker.go

```diff
@@ -54,6 +59,12 @@ func (hec *defaultHecWorker) send(ctx context.Context, bufferState *bufferState,
 	}
 
 	if bufferState.compressionEnabled {
+		r, _ := gzip.NewReader(bytes.NewBuffer(bufferState.buf.Bytes()))
```

Comment:
I wonder if we actually need to pass the bytes to the gzip reader, could we not check the raw bytes directly?

Reply:
Also, make sure you handle any error returned here since that could cause a panic if the ready returned is nil.

Reply:
> I wonder if we actually need to pass the bytes to the gzip reader, could we not check the raw bytes directly?

`bufferState.buf.Bytes()` will be compressed data actually, so we need gzip.NewReader to read the actual data.

Reply:
I don't think we have access to raw bytes once the gzip writer is created.
It will directly write to the compressed buffer.